src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -40,14 +40,20 @@ Copyright (c) .NET Foundation. All rights reserved.
     <IsInnerBuild Condition="'$(TargetFramework)' != '' AND '$(TargetFrameworks)' != ''">true</IsInnerBuild>
     <SymbolPackageFormat Condition="'$(SymbolPackageFormat)' == ''">symbols.nupkg</SymbolPackageFormat>
     <AddPriFileDependsOn Condition="'$(MicrosoftPortableCurrentVersionPropsHasBeenImported)' == 'true'">DeterminePortableBuildCapabilities</AddPriFileDependsOn>
-    <NuspecOutputPath Condition="'$(NuspecOutputPath)' == ''">$(BaseIntermediateOutputPath)$(Configuration)\</NuspecOutputPath>
     <WarnOnPackingNonPackableProject Condition="'$(WarnOnPackingNonPackableProject)' == ''">false</WarnOnPackingNonPackableProject>
     <ImportNuGetBuildTasksPackTargetsFromSdk Condition="'$(ImportNuGetBuildTasksPackTargetsFromSdk)' == ''">false</ImportNuGetBuildTasksPackTargetsFromSdk>
     <AllowedOutputExtensionsInPackageBuildOutputFolder>.dll; .exe; .winmd; .json; .pri; .xml; $(AllowedOutputExtensionsInPackageBuildOutputFolder)</AllowedOutputExtensionsInPackageBuildOutputFolder>
     <AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder Condition="'$(SymbolPackageFormat)' != 'snupkg'">.pdb; .mdb; $(AllowedOutputExtensionsInPackageBuildOutputFolder); $(AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder)</AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>
     <AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder Condition="'$(SymbolPackageFormat)' == 'snupkg'">.pdb</AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>
     <SuppressDependenciesWhenPacking Condition="'$(SuppressDependenciesWhenPacking)' == ''">false</SuppressDependenciesWhenPacking>
   </PropertyGroup>
+
```

Comment:
Add a test in [NuGet/NuGet.Client/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs) to verify the correct behavior.

You can find inspiration from similar tests. For example:

https://github.com/NuGet/NuGet.Client/blob/ca008ff53cd941d31cba2052c1fcf4c31770b294/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs#L4061-L4083

Reply:
OK. Will do.

Reply:
I don't think that'd work.
The problem is that those tests use the SDK and the SDK is the one that sets defaults.

Reply:
I'd look into https://github.com/NuGet/NuGet.Client/tree/dev/test/NuGet.Core.FuncTests/Msbuild.Integration.Test and see if it's at all possible there.

Reply:
> SDK is the one that sets defaults

Do we want the SDK to set the defaults? Both `RestoreOutputPath` and `NuSpecOutputPath` have defaults set in NuGet targets.

> I'd look into [MSBuild.Integration.Test](/NuGet/NuGet.Client/tree/dev/test/NuGet.Core.FuncTests/Msbuild.Integration.Test) and see if it's at all possible there.

There's a PR already on the SDK side. I have tested locally with only pack related tests. They all pass. I recommend to merge this first, since the SDK depends on it.

> I don't think that'd work.

I could add a simple test that fails if the defaults had not been set at the end of the first pass.

Reply:
> Do we want the SDK to set the defaults? Both RestoreOutputPath and NuSpecOutputPath have defaults set in NuGet targets.

We don't need them to.
No one should really use RestoreOutputPath, it's not consistently interpreted, it's MSBuildProjectExtensionsPath that matters.

Reply:
> There's a PR already on the SDK side. I have tested locally with only pack related tests. They all pass. I recommend to merge this first, since the SDK depends on it.

Yeah we'll merge this one first. It doesn't harm us to have the same logic in 2 places.

> I could add a simple test that fails if the defaults had not been set at the end of the first pass.

How would you achieve that if the targets set the defaults themselves?

Reply:
> We don't need them to

Thanks for clarification.

> How would you achieve that if the targets set the defaults themselves?

SDK tests reports error if the defaults not set. I have a project that repros the issue. I have to debug those tests to find the cause.

Reply:
Just do a best effort and we'll try to help if you get stuck. :)